### PR TITLE
Remove DynaLoader, "use vars" and lazy load Carp

### PR DESCRIPTION
**Pull request by atoomic**
Three different changes in this pull request, each of them having their own commit.

* Switch Filter::Util::Call to XSLoader
* Lazy load Carp only when require 
* Replace "use vars" by our

Let me know if you prefer individual pull request for them.

### CODE REVIEW

**Comment by atoomic** [comment]
@rurban from what I can see the travis build errors are not related to these changes and come from missing perl images:
```
unable to find perl for 5.24-mb!
``

**Comment by rurban** [comment]
LGTM

File: Call/Call.pm [hunk 1]
```diff
@@ -10,13 +10,13 @@
 package Filter::Util::Call ;
 
 require 5.005 ;
```
Comment (rurban) [inline Call/Call.pm]: This needs to change to 5.006

File: Call/Call.pm [hunk 2]
```diff
@@ -60,7 +60,7 @@ sub filter_add($)
     Filter::Util::Call::real_import($obj, (caller)[0], $coderef) ;
 }
 
-bootstrap Filter::Util::Call ;
+XSLoader::load( 'Filter::Util::Call', $VERSION );
```
Comment (rurban) [inline Call/Call.pm]: Wrong. Either $XS_VERSION or better no argument at all. XSLoader will use $XS_VERSION.
It only matters with dev releases, but still.

**Comment by atoomic** [comment]
ok I see you manually adjusted the original PR
we can close it, as it's now merged

**Pull request closed without merging**